contributing.md: Add guideline for creating documentation

philipanda opened the pull request.

philipanda committed:
contributing.md: Add  KW documentation guideline

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

contributing.md: docs example: use proper syntax for sections

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

docs/contributing: Fix broken interline spacing in example doc lists

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

docs/contributing.md: Add link to robot docs formatting

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

contributing.md: Add a note that RF 5.0 is used

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

contributing.md: KW docs requirements: split to public and private KWds

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

contributing.md: Fix example to return as documented

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
docs/contributing.md: Add link to robot docs formatting

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

mkopec commented on docs/contributing.md:
```diff
@@ -88,3 +88,52 @@ and add as guidelines:
+    ...    - ``string`` - The generated cowsay message
+    ...
+    ...    === Side Effects ===
+    ...    - Creates ``${out_file}`` file with the cowsay message. Overwrites
```
Comment from https://github.com/Dasharo/open-source-firmware-validation/pull/664#discussion_r1914771368 applies here too. Maybe split this into `Effect` and `Side effects`?

EduKav1813 replied:
We can also just have `Effects`, which would contain both main and side ones, and remove the need to debate which belong where ;)

mkopec replied:
Works for me :)

mkopec replied:
Marking as resolved, this isn't really a blocker tbh

philipanda replied:
https://github.com/Dasharo/open-source-firmware-validation/compare/fd1005c..3de6568

m-iwanicki commented on docs/contributing.md:
```diff
@@ -88,3 +88,75 @@ and add as guidelines:
+    ...    \ path. Does not have to be absolute.
+    ...
+    ...    === Return Value ===
+    ...    - ``string`` - The generated cowsay message
```
In your example we don't return anything

philipanda replied:
That's a great point :smile:
https://github.com/Dasharo/open-source-firmware-validation/compare/76501ef..39152d

philipanda committed:
kaywords.robot: Add "Abstract" Power On keyword to override in configs

Defining "Power On" in platform configs will override it. Having this
"abstract" keyword allows to create a documentation in one place, to
make it appear in the HTML documentation and to be detected by LSP
servers

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
platform-configs: Update Power On impls docs

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options: Document keywords

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
robocop.toml: Increase max kwd length to allow comprehensive docs

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/bios/menus.robot: Document kwds up to the experimental kwds marker

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/bios/menus.robot: Document kwds up to `Exit From Current Menu`

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/bios/menus.robot: Document remainging kwds

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/bios/menus.robot: Remove duplicate&unused `Save Bios Changes`

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/bios/menus.robot: Move one-suite specific kwd to the suite

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/: Add missing resources and libraries imports

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options: Docs: note that the implementations behave differently

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options: Rename impl files to fix duplicate lib names in docs

`lib/options/dcu.robot` and `lib/dcu.robot keywords` all had the prefix
`dcu.`

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
docs/contributing.md: Documentation header `Side Effects` -> `Effects`

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
keyword docs: Rename section `Side Effects` to `Effects`

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/terminal.robot: Add documentation

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/terminal.robot: Execute UEFI Shell Command: disambiguate a var name

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>
